http_proxy.go

```diff
@@ -86,6 +87,8 @@ type HTTPProxyConfig struct {
 	CloseAfterReply        bool
 	DenyDomains            []*regexp.Regexp
 	DenyDomainsExclude     []*regexp.Regexp
+	ReadLimit              uint
```

Comment:
I think it would be beneficial to use a special type that supports kb mb etc, see in rclone for inspiration.